synapse/storage/util/id_generators.py

```diff
@@ -87,7 +89,25 @@ def _load_current_id(
     return (max if step > 0 else min)(current_id, step)
 
 
-class StreamIdGenerator:
+class AbstractStreamIdGenerator(metaclass=abc.ABCMeta):
```

Comment:
Not sure how I feel about this, but it seems useful to have a shared super-class for `StreamIdGenerator` and `MultiWriterIdGenerator`?

Reply:
It's definitely more convenient for type hinting than writing `Union[StreamIdGenerator, MultiWriterIdGenerator]` everywhere

Reply:
> It's definitely more convenient for type hinting than writing `Union[StreamIdGenerator, MultiWriterIdGenerator]` everywhere

Oh I should see where we do that an update it I guess...

Reply:
I couldn't find any instances of this, although there's some of `Union[StreamIdGenerator, SlavedIdTracker]`.